docs/milestones/2.md

```diff
@@ -0,0 +1,89 @@
+
+![](./architecture.png)
+
+[Source](https://whimsical.com/snapshot-x-milestone-2-7UJARrFFC3LbnZexGEtQQB)
+
+### Single Slot Proof Voting Strategy Contract
+
+We add the principal voting strategy of Snapshot X to this milestone. The single slot proof strategy will allow the balances of the majority [2] of ERC20 and ERC721 tokens to be used to find voting power. 
+
+#### Functions
+
+- **get_voting_power(address: EthAddress, ethereum_block_number: felt, params: felt array)**  
+Returns the voting power for a specific address at a specific Ethereum block number **block_number**, **params** are the storage proof data required by the Fossil module to prove the contents of a slot. In the case of finding the voting power from an ERC20 balance, the slot will correspond to the storage slot in the ERC20 contract where the balance of **address** is stored. 
+
+### StarkNet Commit Contract
+
+This global contract will allow an alternate way to interact with Snapshot X: Via an L1 transaction. The core benefit of this feature is to allow one to interact with Snapshot X without a signature. This will give smart contract accounts such as multi-sig wallets the ability to participate. 
+
+The contract will have a single storage variable: The address of the global L1 Auth contract on StarkNet. This will be an immutable variable set at construction of the contract.
+
+#### Functions 
+
+- **commit(hash: bytes32)** \
+Sends the hash provided to the StarkNet messaging bridge with the L1 Auth contract as the destination address.
+
+### L1 Auth Contract 
+
+Auth contract for interactions with Snapshot X via L1. 
+
+#### Functions 
+
+- **commit_handler(hash: felt array)** \
+L1 handler function that receives the hash sent by the Starknet Commit contract and stores it in its state.
+
+- **execute(to : felt, function_selector : felt, calldata : felt array)** \
+General purpose athentication for all types of interaction where the interaction parameters are sent and hashed to check that they match a hash recieved from L1. If so, the function given by **function_selector** at the **to** address is called with **calldata** as parameters. 
+
+### Zodiac Execution Strategy Contract
+
+TODO 
+
+### Zodiac Module Contract
+
+A way to execute Gnosis Safe transactions that are included in a proposal, once the vote has concluded successfully.
+
+#### Functions
+
+**receiveProposal(executionDetails: Uint256, hasPassed: Uint256, txHashes: bytes32[])** \
+Consumes message from StarkNet bridge, checks proposal has passed, and then hashes the **txHashes** array to check it is equal to **executionDetails**. If so, it stores the **txHashes** ready to be executed in a following transaction.
+
+**executeProposalTxBatch(proposalIndex: Uint256, Raw Tx parameter arrays ...)** \
+Hashes each transaction in turn and checks that it is equal to the corresponding txHash stored. If correct, executes that transaction. 
```

Comment:
what's the use of `proposalIndex` ? I don't think the function `receiveProposal` mentions it, so is it a variable that's internal to the Zodiac Module Contract? It's not the same `proposalIndex` as the one on the L2 space contract right?

Reply:
Yeah its just a counter that is incremented each time a new proposal is received, its used to access the proposals in the contract so works like an ID. I called it `proposalIndex` rather than `proposalID` so that it was not mixed up with the L2 one. maybe we should use a different name all together for clarity. `proposalNumber`?